tests-integration/timeline_test.go

```diff
@@ -689,6 +690,120 @@ func TestTimelineTxnID(t *testing.T) {
 	))
 }
 
+// Like TestTimelineTxnID, but where ...
+func TestTimelineTxnIDAfterInitialSync(t *testing.T) {
```

Comment:
I'm not actually sure. The only significant difference I can spot is 

- the initial poll responses have a next_batch token. I vaugely remember that the pollers were confused if they didn't have since token from a previous iteration
- the original test didn't interrogate what Alice sees if her sync request is served before her poller sees the txn_id.

I think the latter bullet is the crucial bit. Will amend.